### DISCUSSION TITLE
Multiple target update

### DISCUSSION THREAD

**Discussion by sarunac** [karimra/gnmic#354] [Q&A]
Is there a way to update multiple targets with individual update values at the same time ?
For example: Can I achieve this in any way ?

gnmic -a <ADDRESS1>  --insecure set --update-path  PATH1 --update-file FILE1  -a <ADDRESS2> set --update-path  PATH1 --update-file FILE2

**Comment by karimra** [comment]
I'm afraid that not possible, 
and not really possible to implement with the current libraries.

I would use a simple bash script like this:
```bash
#!/bin/bash

set -e

ip_addrs=(
    "172.1.2.3"
    "191.1.2.3"
)
update_file_names=(
    "/path/to/file1"
    "/path/to/file2"
)

for i in "${!ip_addrs[@]}"
do
    gnmic -a "${ip_addrs[$i]}" --insecure set --update-path $PATH1 --update-file "${update_file_names[$i]}"
done
```

EDIT:
What would be doable as a new feature is have the files evaluated as Go templates, 
add a flag --env-file that points to a file containing the input to render the templates.
That way a single file (template) is rendered based on each target (router) variables.
